test/specs/core.controller.tests.js

```diff
@@ -1447,20 +1452,72 @@ describe('Chart', function() {
 				chart.destroy();
 
 				expect(sequence).toEqual([].concat(
-					hooks.init,
+					hooks.install,
+					hooks.enable,
```

Comment:
I'm wondering if it makes sense to call `enable` if the state didn't change because the plugin already knows from `install` that it's initially enabled/disabled. Would you call `disable` here as well if the plugin is initially disabled?

What do you think?

Reply:
Thats actually why I called it `enabled`
If you disconnect enabled/disable from install/uninstall, it makes sense. And disconnecting those makes sense for simpler plugin state management.

Essentially avoiding the need for same code in install/enable in a plugin that does not care about the Chart init or destroy

Reply:
This is debatable since it's easy to refactor common code between install/enable and call a single function from both. But you may also want a different logic when initially enabled vs the state changes from disabled to enabled.

I don't think the name `enabled` makes it more obvious: It would be `onEnabled` which I think means that the state has effectively changed (similar to the `onStateChanged` suggested by @etimberg).

If we keep calling `enable` here, then we also need to call `disable` for consistency if the plugin is initially disabled.

Reply:
IMO one of the important goals of a plugin interface is to make things easy and simple. The proposed way supports 2 new ways of "state handling" without actual state handling in the plugin.

> But you may also want a different logic when initially enabled vs the state changes from disabled to enabled, which is currently harder to detect.

I don't think there are any existing plugins that do such a thing (because it is impossible to detect disabling before this). 
If you need to detect the state change, it means you have some internal state handling. Thus, how is it harder to detect?

```js
{
  install(chart, args, options) {
    // always called
    // state = options !== false
  },
  beforeInit(chart) {
    // only called if enabled
  },
  enable(chart) {
    // only called if enabled
    // changed = state === false
    // state = true
  },
  disable(chart) {
    // only called if enabled
    // state = false
  },
  destroy(chart) {
    // only called if enabled
  },
  uninstall(chart) {
    // always called
  }
}
```

Can you describe a use case that is harder to do with these vs omitting the first enabled call on an initially enabled plugin?

Reply:
Ok, I think you are right, maybe `start/stop` would be better names than `enable/disable` then?

When I read `enable`, I expect it to be called if the previous state was "disabled" (which is not necessary the case after `install`). But I would expect `start` to be called after install if the plugin is initially enabled or has been re-enabled.

`install` > `start` (if enabled) > `stop` (if previously started) > `start` > `uninstall`